test/00-init.ts

```diff
@@ -0,0 +1,49 @@
+import * as Promise from 'bluebird';
+import * as _ from 'lodash';
+import * as fs from 'fs';
+import * as path from 'path';
+
+const testFiles = _(process.env.TEST_FILES)
+	.trim()
+	.split(' ')
+	.map(
+		(fileName): ((f: string) => boolean) => {
+			const [op, ...rest] = fileName;
+			const compareFilename = rest.join('');
+			switch (op) {
+				case '<':
+					return f => f < compareFilename;
+				case '>':
+					return f => f > compareFilename;
+				default:
+					return f => _.startsWith(f, fileName);
+			}
+		},
+	);
+
+const prefixes: Dictionary<true> = {};
+
+Promise.resolve(fs.readdirSync(__dirname))
```

Comment:
This can be
```ts
Promise.resolve(fs.promises.readdir(__dirname))
```
to avoid the sync call

Reply:
@Page- my types don't have `.promises.readdir`

Reply:
I updated `"@types/node": "^10.14.4"` to provide the extra typing